src/NServiceBus.Persistence.AzureStorage/SagaPersisters/AzureSagaPersister.cs

```diff
@@ -76,19 +76,33 @@ public async Task Complete(IContainSagaData sagaData, SynchronizedStorageSession
         {
             var table = await GetTable(sagaData.GetType()).ConfigureAwait(false);
 
-            var sagaId = sagaData.Id;
-            var query = GenerateSagaTableQuery<DictionaryTableEntity>(sagaId);
-
-            var entity = (await table.ExecuteQueryAsync(query).ConfigureAwait(false)).SafeFirstOrDefault();
-            if (entity == null)
+            var sagaId = sagaData.Id.ToString();
+            var meta = context.GetOrCreate<SagaInstanceMetadata>();
+            if (!meta.TryGetEtag(sagaData, out var etag))
             {
-                return; // should not try to delete saga data that does not exist, this situation can occur on retry or parallel execution
+                etag = "*";
             }
 
-            await table.DeleteIgnoringNotFound(entity).ConfigureAwait(false);
+            var entity = new DictionaryTableEntity
+            {
+                ETag = etag,
+                PartitionKey = sagaId,
+                RowKey = sagaId
+            };
+            try
+            {
+                await table.ExecuteAsync(TableOperation.Delete(entity)).ConfigureAwait(false);
+            }
+            catch (StorageException e)
+            {
+                if (e.RequestInformation.HttpStatusCode != 404)
```

Comment:
@SeanFeldman I've reintroduced this behavior because there are tests in place that verify it behaves that way so therefore I guess we can assume the persister is expected to behave like this when we try to delete but can't find the record anymore. Without this the following tests would fail

- When_completing_saga.Should_allow_action_twice_without_throwing_error
- When_completing_saga.Should_succeed_if_saga_doesnt_exist